src/crowdin_gui.cpp

```diff
@@ -676,6 +677,28 @@ bool ExtractCrowdinMetadata(CatalogPtr cat,
                 : cat->GetLanguage();
     }
 
+    if (const auto xliff = dynamic_cast<XLIFFCatalog*>(cat.get()))
```

Comment:
`std::dynamic_pointer_cast<>` is your friend.

Also, please don't use this sort of valid-but-cryptic constructs like assignment within `if` with reliance on implicit null check.

Reply:
Corrected https://github.com/vslavik/poedit/pull/644/commits/87e824ed0fa0be42e13e46847bfc1119a645ad06#diff-a1865a3025926eb82afbd7f4d9f13518R680